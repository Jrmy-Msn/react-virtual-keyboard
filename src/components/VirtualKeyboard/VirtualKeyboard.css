.VirtualKeyboard[theme="dark"] {
  --primary: #292929;
  --primary-light: #a9a9a9;
  --primary-lighter: #f2f2f2;
  --primary-dark: #212121;
  --primary-darker: #141414;
  --primary-text-light: #fff;
  --primary-text-dark: #000;
  --disabled-text: #949494;
}

.VirtualKeyboard[theme="light"] {
  --primary: #f2f2f2;
  --primary-light: #515151;
  --primary-lighter: #212121;
  --primary-dark: #fefefe;
  --primary-darker: #fff;
  --primary-text-light: #000;
  --primary-text-dark: #fff;
  --disabled-text: #949494;
}

.VirtualKeyboard {
  --letter-width: 8%;

  display: flex;
  flex-direction: column;
  background-color: var(--primary);
}

.virtual-keyboard-letters {
  display: flex;
  flex-direction: column;
}

.virtual-keyboard-row {
  display: flex;
  justify-content: center;
}

.virtual-keyboard-action {
  height: 1.5rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
  cursor: pointer;
  background-color: var(--primary-light);
}

.virtual-keyboard-action > i {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  flex-grow: 1;
  height: 100%;
  text-align: center;
  color: var(--primary-text-dark)
}

.virtual-keyboard-action > i:hover {
  flex-grow: 1;
  text-align: center;
  background-color: var(--primary-lighter);
}