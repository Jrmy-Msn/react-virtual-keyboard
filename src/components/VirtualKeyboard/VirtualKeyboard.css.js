export const usedStyle = theme => ({
  root: {
    borderRadius: 0,
  },
  key: {
    flexGrow: 1
  }
})