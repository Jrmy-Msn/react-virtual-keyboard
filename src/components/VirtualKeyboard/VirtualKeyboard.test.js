import React from 'react'
import {cleanup, fireEvent, render, screen} from '@testing-library/react'

import VirtualKeyboard from "./VirtualKeyboard"

afterEach(cleanup)

it('should display 26 <VirtualKey /> component', () => {
  render(<VirtualKeyboard />)
  const aKeys = screen.getAllByRole('button', {name: /Lettre [A-Z]/})
  expect(aKeys).toHaveLength(26)
})

it('should display X <VirtualKey /> component based on number of letter in "layout prop', () => {
  render(<VirtualKeyboard layout={['JRMY', 'MSN']} />)
  const aKeys = screen.getAllByRole('button', {name: /Lettre [A-Z]/})
  expect(aKeys).toHaveLength(7)
})

it('should <VirtualKey /> component with "currentKey" prop as "value" prop have a "feedback" prop', () => {
  const feedbackForCurrentKey = jest.fn()
  feedbackForCurrentKey.mockImplementation((p) => p === 'J' ? 'pressed' : '')
  render(<VirtualKeyboard currentKey={'J'} feedbackForCurrentKey={feedbackForCurrentKey} />)
  expect(screen.getByText('J')).toHaveClass('pressed')
})

it('should call "onKeyDown" prop after start pressed a physical key', () => {
  const onKeyDown = jest.fn()
  render(<VirtualKeyboard onKeyDown={onKeyDown}/>)
  fireEvent.keyDown(screen.getByLabelText('Clavier virtuel'))
  expect(onKeyDown).toHaveBeenCalledTimes(1)
})

it('should call "onKeyUp" prop after finish pressed a physical key', () => {
  const onKeyUp = jest.fn()
  render(<VirtualKeyboard onKeyUp={onKeyUp}/>)

  fireEvent.keyUp(screen.getByLabelText('Clavier virtuel'))
  expect(onKeyUp).toHaveBeenCalledTimes(1)
})

it('should call "feedbackForCurrentKey" prop for each <VirtualKey /> component', () => {
  const feedbackForCurrentKey = jest.fn()
  feedbackForCurrentKey.mockImplementation(() => 'pressed')
  render(<VirtualKeyboard feedbackForCurrentKey={feedbackForCurrentKey}/>)
  const aKeys = screen.getAllByRole('button', {name: /Lettre [A-Z]/})
  expect(feedbackForCurrentKey).toHaveBeenCalledTimes(aKeys.length)
})

