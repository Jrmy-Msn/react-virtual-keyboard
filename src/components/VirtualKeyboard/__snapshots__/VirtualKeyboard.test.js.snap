// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualKeyboard Component Render renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    aria-label="clavier"
    aria-owns=".VirtualKey"
    className="MuiGrid-root VirtualKeyboard MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        aria-label="disposition"
        className="MuiButtonBase-root MuiButton-root VirtualKeyboard-root-1 MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeSmall"
          >
            <i
              aria-hidden="true"
              className="la la-keyboard la-lg"
              role="presentation"
            />
          </span>
          Disposition
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
    >
      <div
        className="MuiButtonGroup-root MuiButtonGroup-disableElevation MuiButtonGroup-contained"
        role="group"
      >
        <button
          aria-label="A"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-5 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            A
          </span>
        </button>
        <button
          aria-label="B"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-6 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            B
          </span>
        </button>
        <button
          aria-label="C"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-7 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            C
          </span>
        </button>
        <button
          aria-label="D"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-8 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            D
          </span>
        </button>
        <button
          aria-label="E"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-9 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            E
          </span>
        </button>
        <button
          aria-label="F"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-10 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            F
          </span>
        </button>
        <button
          aria-label="G"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-11 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            G
          </span>
        </button>
        <button
          aria-label="H"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-12 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            H
          </span>
        </button>
        <button
          aria-label="I"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-13 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            I
          </span>
        </button>
        <button
          aria-label="J"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-14 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            J
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
    >
      <div
        className="MuiButtonGroup-root MuiButtonGroup-disableElevation MuiButtonGroup-contained"
        role="group"
      >
        <button
          aria-label="K"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-15 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            K
          </span>
        </button>
        <button
          aria-label="L"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-16 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            L
          </span>
        </button>
        <button
          aria-label="M"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-17 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            M
          </span>
        </button>
        <button
          aria-label="N"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-18 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            N
          </span>
        </button>
        <button
          aria-label="O"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-19 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            O
          </span>
        </button>
        <button
          aria-label="P"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-20 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            P
          </span>
        </button>
        <button
          aria-label="Q"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-21 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Q
          </span>
        </button>
        <button
          aria-label="R"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-22 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            R
          </span>
        </button>
        <button
          aria-label="S"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-23 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            S
          </span>
        </button>
        <button
          aria-label="T"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-24 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            T
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
    >
      <div
        className="MuiButtonGroup-root MuiButtonGroup-disableElevation MuiButtonGroup-contained"
        role="group"
      >
        <button
          aria-label="U"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-25 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            U
          </span>
        </button>
        <button
          aria-label="V"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-26 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            V
          </span>
        </button>
        <button
          aria-label="W"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-27 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            W
          </span>
        </button>
        <button
          aria-label="X"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-28 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            X
          </span>
        </button>
        <button
          aria-label="Y"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-29 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Y
          </span>
        </button>
        <button
          aria-label="Z"
          className="MuiButtonBase-root MuiButton-root makeStyles-root-3 MuiButton-contained VirtualKey MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-active-4 makeStyles-active-30 MuiButton-containedPrimary MuiButton-disableElevation"
          disabled={false}
          feedback=""
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Z
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
